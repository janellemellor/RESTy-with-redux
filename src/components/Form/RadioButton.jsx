import React from 'react';
import PropTypes from 'prop-types';

export const RadioButtonGroup = ({ children, name, onChange }) => {
  const radioButtons = React.Children.map(children, child => {
    return React.cloneElement(child, {
      name, 
      onChange
    });
  });

  return (
    <>
      {radioButtons}
    </>
  );
};

export const RadioButton = ({ name, value, onChange }) => (
  <>
    <input type ="radio" id={value} name={name} value={value} onChange={onChange} />
    <label htmlFor={value}>{value}</label>
  </>
);

RadioButtonGroup.propTypes = {
  children: PropTypes.node, 
  name: PropTypes.string.isRequired, 
  onChange: PropTypes.func.isRequired
};

RadioButton.propTypes = {
  name: PropTypes.string.isRequired, 
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string
};
