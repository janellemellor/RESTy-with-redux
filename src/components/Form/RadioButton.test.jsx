import React from 'react';
import { shallow } from 'enzyme';
import { RadioButton } from './RadioButton';

describe('RadioButton component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<RadioButton id="GET" name="methods" value="GET" onChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
