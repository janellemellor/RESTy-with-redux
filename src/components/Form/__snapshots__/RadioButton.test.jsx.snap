// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component matches a snapshot 1`] = `
<Fragment>
  <input
    id="GET"
    name="methods"
    onChange={[Function]}
    type="radio"
    value="GET"
  />
  <label
    htmlFor="GET"
  >
    GET
  </label>
</Fragment>
`;
